.github/workflows/build.yml

```diff
@@ -0,0 +1,282 @@
+              run: |
+                  mkdir -p artifacts
+                  cp -r ./service/build/reports ./artifacts
+
+            - name: Upload Artifacts
+              uses: actions/upload-artifact@v4
+              if: always()
+              with:
+                  name: 'Spotbugs Reports'
+                  path: ./artifacts/**
+                  compression-level: 9
+
+            - name: Temporary Artifacts Cleanup
+              run: rm -rf ./artifacts
+
+    unit-tests:
+        name: Unit Tests
+        runs-on: ubuntu-latest
+        needs: [ checkstyle, spotbugs ]
+        steps:
+            -   name: Checkout Repository
+                uses: actions/checkout@v4
+                with:
+                    fetch-depth: 0
+            -   name: Setup Java 21 LTS
+                uses: actions/setup-java@v4
+                with:
+                    java-version: 21
+                    distribution: 'temurin'
+
+            -   name: Execute Unit Tests
+                run: ./gradlew test jacocoTestReport --parallel --build-cache
+                working-directory: ./service
+
+            -   name: Collect Artifacts
+                if: always()
+                run: |
+                    mkdir -p artifacts
+                    cp -r ./service/build/reports ./artifacts
+
+            -   name: Upload Artifacts
+                uses: actions/upload-artifact@v4
+                if: always()
+                with:
+                    name: 'Unit Test Reports'
+                    path: ./artifacts/**
+                    compression-level: 9
+
+            -   name: Temporary Artifacts Cleanup
+                run: rm -rf ./artifacts
+
+    integration-tests:
+        name: Integration Tests
+        runs-on: ubuntu-latest
+        needs: [ checkstyle, spotbugs ]
+        steps:
+            -   name: Checkout Repository
+                uses: actions/checkout@v4
+
+            -   name: Setup Java 21 LTS
+                uses: actions/setup-java@v4
+                with:
+                    java-version: 21
+                    distribution: 'temurin'
+
+            -   name: Start Docker Dependencies
+                env:
+                    PEM111_AMQP_BROKER: "amqp://activemq:5672"
+                    PEM111_AMQP_PORT: "5672"
+                    PEM111_ITK_EXTERNAL_CONFIGURATION_URL: "http://wiremock:8080/configuration/ods-dos"
+                    LOG_LEVEL: DEBUG
+                run: |
+                    docker network create 111network
+                    docker compose build
+                    docker compose up activemq wiremock --detach
+                working-directory: ./docker
+
+            -   name: Execute Integration Tests
+                run: ./gradlew integrationTest
+                working-directory: ./service
+
+            -   name: Dump Docker Logs
+                if: always()
+                run: |
+                    chmod +x dump_docker_logs.sh
+                    ./dump_docker_logs.sh
```

Comment:
Why not commit the file with the execute bit set, thus avoiding needing this?

Reply:
I tried but it doesn't work hence will keep it in place

Reply:
I tried, and it does work in #313